docs/develop/typescript/message-passing.mdx

```diff
@@ -196,107 +290,319 @@ await client.workflow.signalWithStart(yourWorkflow, {
-Queries are read-only and must complete synchronously.
-A Query handler may not return a Promise.
+An Update is a synchronous, blocking call that can change Workflow state, control its flow, and return a result.
 
-Queries are typically used to access the state of an open (running) Workflow Execution.
-It is also possible to send a Query to a closed Workflow Execution.
-In both cases, you must have access to at least one running Worker for the Task Queue that the Workflow belongs to.
+A client sending an Update must wait until the Server delivers the Update to a Worker.
+Workers must be available and responsive.
+If you need a response as soon as the Server receives the request, use a Signal instead.
+Also note that you can't send Updates to other Workflow Executions or perform an Update equivalent of Signal-With-Start.
 
-The Query is identified at both ends by a Query Type.
-The Workflow requires a Query handler to process the Query and provide data representing the requested information.
-Queries return the most recent state of the Workflow Execution.
-The returned data reflects the state of all confirmed Events before the Query was sent.
-An Event is confirmed if the call creating the Event returned a success.
-If an Event is created while the Query is underway, it may not be reflected in the Query results.
+- `WorkflowExecutionUpdateAccepted` is added to the Event History when the Worker confirms that the Update passed validation.
+- `WorkflowExecutionUpdateCompleted` is added to the Event History when the Worker confirms that the Update has finished.
 
-### Define a Query {#define-query}
+To send an Update to a Workflow Execution, you can:
 
-A Query has a name and can have arguments.
+- Call [`WorkflowHandle.executeUpdate`](https://typescript.temporal.io/api/interfaces/client.WorkflowHandle/#executeUpdate) and wait for the Update to complete.
+  This code fetches an Update result:
 
-- The name, also called a Query type, is a string.
-- The arguments must be [serializable](/dataconversion).
+  ```typescript
+  let previousLanguage = await handle.executeUpdate(setLanguage, {
+    args: [Language.CHINESE],
+  });
+  ```
 
-Use [`defineQuery`](https://typescript.temporal.io/api/namespaces/workflow/#definequery) to define the name, parameters, and return value of a Query.
-This returns a `QueryDefinition` object.
-It carries the Query's signature.
-This includes the Query Type, the input parameters types, and the Query's return type.
+- Send [`WorkflowHandle.startUpdate`](https://typescript.temporal.io/api/interfaces/client.WorkflowHandle/#startUpdate) to receive an [`WorkflowUpdateHandle`](https://typescript.temporal.io/api/interfaces/client.WorkflowUpdateHandle) as soon as the Update is accepted or rejected.
 
-<!--SNIPSTART typescript-define-query -->
+  - Use this `UpdateHandle` later to fetch your results.
+  - `async` Update handlers normally perform long-running async activities.
```

Comment:
> normally perform long-running async activities.

"long-running activities" is confusing; sounds like that's related with Activities.

Reply:
Thanks, fixed here and in Python.